tests/protocol_muxer/test_protocol_muxer.py

```diff
@@ -0,0 +1,88 @@
+import pytest
```

Comment:
Really nice test setup. Only thing to add is that we should have a better way of assigning the ports used in tests, maybe by a _next_port function that gives back a monotonically increasing port number (this seems like a bit of a hack, eventually we probably want to mock the ports). Will open an issue for that later.